src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs

```diff
@@ -122,22 +122,46 @@ public static SubscriptionClient Create(ServiceBusEntityConnection topicConnecti
             await this.OnCloseAsync().ConfigureAwait(false);
         }
 
-        public async Task<BrokeredMessage> ReceiveAsync()
+        /// <summary>
+        /// Asynchronously receives a message using the <see cref="MessageReceiver" />.
+        /// </summary>
+        /// <returns>The asynchronous operation.</returns>
+        public Task<BrokeredMessage> ReceiveAsync()
         {
-            IList<BrokeredMessage> messages = await this.ReceiveAsync(1).ConfigureAwait(false);
-            if (messages != null && messages.Count > 0)
-            {
-                return messages[0];
-            }
+            return this.InnerReceiver.ReceiveAsync();
+        }
 
-            return null;
+        /// <summary>
+        /// Asynchronously receives a message using the <see cref="MessageReceiver" />.
+        /// </summary>
+        /// <param name="serverWaitTime">The time span the server waits for receiving a message before it times out.</param>
+        /// <returns>The asynchronous operation.</returns>
+        public Task<BrokeredMessage> ReceiveAsync(TimeSpan serverWaitTime)
+        {
+            return this.InnerReceiver.ReceiveAsync(serverWaitTime);
         }
 
+        /// <summary>
+        /// Asynchronously receives a message using the <see cref="MessageReceiver" />.
+        /// </summary>
+        /// <param name="maxMessageCount">The maximum number of messages that will be received.</param>
+        /// <returns>The asynchronous operation.</returns>
         public Task<IList<BrokeredMessage>> ReceiveAsync(int maxMessageCount)
         {
             return this.InnerReceiver.ReceiveAsync(maxMessageCount);
         }
 
+        /// <summary>
+        /// Asynchronously receives a message using the <see cref="MessageReceiver" />.
```

Comment:
More of a question - "Asynchronously receives a message". Is "Asynchronously" really needed? The method is suffixed with `Async` and it returns a `Task` (or `Task<T>`). 😃